src/main/java/org/jenkinsci/plugins/ewm/steps/ExwsExecution.java

```diff
@@ -67,8 +79,104 @@ public boolean start() throws Exception {
+        }
+    }
+
+    /**
+     * Finds a template from the given templates list that matches the given label String.
+     *
+     * @param diskPoolRefId   the disk pool ref id that the found template should have
+     * @param nodeLabelString the label that the template has
+     * @param templates       a list of templates
+     * @return the template matching the given label, <code>null</code> otherwise
+     * @throws IOException if the found template doesn't have defined a disk pool ref id
+     *                     if the defined disk pool ref id doesn't match the one given as parameter
+     */
+    @CheckForNull
+    private static Template findTemplate(String diskPoolRefId, String nodeLabelString, List<Template> templates) throws IOException {
+        Template selectedTemplate = null;
+        for (Template template : templates) {
+            String templateLabel = template.getLabel();
+            if (templateLabel != null && nodeLabelString.contains(templateLabel)) {
+                selectedTemplate = template;
+                break;
+            }
+        }
+        if (selectedTemplate != null) {
+            String templateDiskPoolRefId = selectedTemplate.getDiskPoolRefId();
+            if (templateDiskPoolRefId == null) {
+                throw new AbortException("The Disk Pool Ref ID was not provided in Jenkins global config");
+            }
+            if (!templateDiskPoolRefId.equals(diskPoolRefId)) {
+                throw new AbortException("The Disk Pool Ref ID defined in Jenkins global config does not match the one allocated by the exwsAllocate step");
+            }
+        }
+
+        return selectedTemplate;
+    }
+
+    /**
+     * Finds the {@link NodeProperty} for the external workspace definition.
+     *
+     * @param diskPoolRefId  the disk pool ref id that the node property should have
+     * @param nodeProperties all the the available node properties
+     * @return the node property for the external workspace manager
+     * @throws IOException if no node property was found
+     *                     if the node property doesn't have defined any disk pool ref id
+     *                     if the disk pool ref id doesn't match the given disk pool ref id
+     */
+    @Nonnull
+    private static ExternalWorkspaceProperty findNodeProperty(String diskPoolRefId,
+                                                              DescribableList<NodeProperty<?>, NodePropertyDescriptor> nodeProperties) throws IOException {
+        ExternalWorkspaceProperty exwsNodeProperty = null;
+        for (NodeProperty<?> nodeProperty : nodeProperties) {
+            if (nodeProperty instanceof ExternalWorkspaceProperty) {
+                exwsNodeProperty = (ExternalWorkspaceProperty) nodeProperty;
+                break;
+            }
+        }
+        if (exwsNodeProperty == null) {
+            throw new AbortException("There is no External Workspace config defined in Node config");
+        }
```

Comment:
I think we forgot a use case where all the nodes have the same mounts as the master node (e.g. in a homogeneous network). Consequently, an exception for this condition here may be too strong. I think we need a second fallback position: the global config. So the search order would be: template, node, global. If after looking up the global config nothing is found, then throw an exception. Perhaps there should also be a short message in the console indicating the search phase being run (so up to three messages). WDYT?